prototype/galore/README.md

```diff
@@ -0,0 +1,33 @@
+## Fused GaLore Adam (WIP)
+
+### Various fused implementations of `Adam` update step per [Gradient Low-Rank Projection](https://arxiv.org/abs/2403.03507)
+
+This is an initial attempt at optimizing the update step of the `GaLore Adam` optimizer.
+
+#### Overview
+
+The `GaLore` `Adam` optimizer introduces additional ops to the traditional `adam` update step.
+
+Specifically:
+
+1.  `grad` is projected to low rank --> additional matmul
+2.  `adam` states are updated with `grad` elementwise (same as `Adam` except in low-rank)
+3.  normalized `grad` is projected to full rank --> additional matmul
+4.  `params` are updated with the normalized full rank grad
+
+#### Installation
+
+```
+pip install --editable .
```

Comment:
mentioned this already but we can package prototype under its own namespace in ao as opposed to its own package